src/section_handlers/document_tracking.py

```diff
@@ -0,0 +1,154 @@
+import logging
+import re
+import sys
+
+from typing import Tuple
+from src.common.common import SectionHandler
+from utils import get_utc_timestamp
+from operator import itemgetter
+from contextlib import suppress
+
+
+class DocumentTracking(SectionHandler):
+    def __init__(self, cvrf2csaf_name, cvrf2csaf_version):
+        super().__init__()
+        self.cvrf2csaf_name = cvrf2csaf_name
+        self.cvrf2csaf_version = cvrf2csaf_version
+
+    def _process_mandatory_elements(self, root_element):
+        self.csaf['id'] = root_element.Identification.ID.text
+        self.csaf['current_release_date'] = root_element.CurrentReleaseDate.text
+        self.csaf['initial_release_date'] = root_element.InitialReleaseDate.text
+        self.csaf['status'] = root_element.Status.text
+
+        revision_history, version = self._process_revision_history(root_element)
+        self.csaf['revision_history'] = revision_history
+        self.csaf['version'] = version
+
+        # Generator is set by this Converter
+        self.csaf['generator'] = {}
+        self.csaf['generator']['date'] = get_utc_timestamp()
+        self.csaf['generator']['engine'] = {}
+        self.csaf['generator']['engine']['name'] = self.cvrf2csaf_name
+        self.csaf['generator']['engine']['version'] = self.cvrf2csaf_version
+
+    def _process_optional_elements(self, root_element):
+        if hasattr(root_element.Identification, 'Alias'):
+            aliases = []
+            for alias in root_element.Identification.Alias:
+                aliases.append(alias.text)
+
+            self.csaf['aliases'] = aliases
+
+    @staticmethod
+    def check_for_version_t(revision_history):
+        """
+        Checks whether all version numbers in /document/tracking/revision_history match semantic versioning.
+        semantic version is defined in version_t definition
+        see: https://docs.oasis-open.org/csaf/csaf/v2.0/csd01/csaf-v2.0-csd01.html#3111-version-type and
+        section 9.1.5 Conformance Clause 5: CVRF CSAF converter
+        """
+
+        pattern = (
+            r'^((0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)'
+            r'(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)$'
+        )
+        return all(re.match(pattern, revision['number']) for revision in revision_history)
+
+    @staticmethod
+    def _as_int_tuple(text: str) -> Tuple[int]:
+        """ Converts string of dotted numbers into tuple of ints """
+        try:
+            return tuple(int(part) for part in text.split('.'))
+        except ValueError:
+            return (sys.maxsize, )
+
+    def _add_current_revision_to_history(self, root_element, revision_history) -> None:
```

Comment:
Please see my [comment](https://github.com/csaf-tools/CVRF-CSAF-Converter/pull/21#issuecomment-1004248100) and the suggested implementation. IMHO the current implementation covers only a minor edge case. Therefore, we should make a broader approach.

Reply:
As written in #25